packages/libs/react-ui/src/components/Button/Button.tsx

```diff
@@ -1,43 +1,72 @@
+import { SystemIcon } from '../Icons';
+
 import { colorVariants } from './Button.css';
+import { ButtonIcon } from './ButtonIcon';
 
 import React, { ButtonHTMLAttributes, FC } from 'react';
 
 export interface IButtonProps
   extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'as' | 'disabled'> {
   as?: 'button' | 'a';
-  onClick?: React.MouseEventHandler<HTMLButtonElement>;
-  href?: string;
+  color?: keyof typeof colorVariants;
   children: React.ReactNode;
-  title?: string;
   disabled?: boolean;
-  color?: keyof typeof colorVariants;
+  href?: string;
+  icon?: keyof typeof SystemIcon;
+  iconAlign?: 'left' | 'right';
+  loading?: boolean;
```

Comment:
todo: animate/spin loading icon with some css :)

Reply:
I've implemented it using this example; https://codepen.io/eveness/pen/BjLaoa

Reply:
Ohh I thought of just doing a continuous rotate transform of 0 to 360deg would do it as the image we're using is just a quarter slice of a circle outline ... if that makes sense haha